### PR TITLE
Add final touch to drag and drop handling of attachments

### PR DESCRIPTION
**Pull request by manuschiller**
This adds:
- highlight class on dragover to show the ability to drop files on dragover @HDinger feel free to change the styling if you have any better looking ideas
- show workpackage in details pane after drop of an attachment in wp-table 
- separated code segments of our former drop handler into several methods for better readability
- bugfix for the "whitespace bug": if the user did not enter a whitespace before and after the attachment markup it got not rendered correctly (eg: `this is my text!any-image.png!` => `this is my text !any-image.png!`)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://manuschiller-attachments.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://manuschiller-attachments.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

File: frontend/app/components/work-packages/wp-single-view/wp-single-view.directive.html [hunk 1]
```diff
@@ -102,6 +102,6 @@ <h3 class="attributes-group--header-text" ng-bind="$ctrl.text.idLabel"></h3>
 
   <wp-attachments
       work-package="$ctrl.workPackage"
-      hide-empty-fields="$ctrl.hideEmptyFields">
+      data-ng-show="$ctrl.filesExist()">
```
Comment (oliverguenther) [inline frontend/app/components/work-packages/wp-single-view/wp-single-view.directive.html]: Wouldn't this cause the attachment section to never show initially despite `show all` selected?

Comment (manuschiller) [inline frontend/app/components/work-packages/wp-single-view/wp-single-view.directive.html]: @oliverguenther I worked on the drag and drop feature for so long that i forgot that one can actually drop files to the dropzone as well. so, you are right ;)

**Comment by manuschiller** [comment]
just noticed that i have to adapt furinvaders changes regarding the retrieval of attachments, so this is again wip

**Comment by manuschiller** [comment]
just checked in on teatro and it looks good to me.
ready for review

**Comment by oliverguenther** [comment]
LGTM 👍  I'd like to have the drag/drop icon removed on items that aren't really draggable but it's not really necessary in this PR.

anim

File: frontend/app/components/work-packages/wp-attachments-formattable-field/wp-attachments-formattable.directive.ts [hunk 2]
```diff
@@ -139,7 +191,7 @@ function wpAttachmentsFormattable() {
       if (angular.isUndefined(controllers[0] && angular.isUndefined(controllers[1]))) {
         return;
       }
-
+      console.log(controllers);
```
Comment (oliverguenther) [inline frontend/app/components/work-packages/wp-attachments-formattable-field/wp-attachments-formattable.directive.ts:194]: Superfluous log line. It's the only remark I have on the PR, so I'll remove it after merging.

**Pull request merged**